### DISCUSSION TITLE
[Proposal] Async Foreach Prefetching

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/runtime#65266] [Ideas] [answered]
Inside a typical async foreach loop
```C#
await foreach(var ele in MyAsyncEnumerable())
{
    //do something
}
```
if it's safe to assume that the enumeration of ele doesn't depend on what happens inside the body of the loop, for example reading a file over ftp, we can start doing that (prefetching) for the next enumeration while the current enumeration's body in the for loop is still executing.

**Comment by Joe4evr** [comment, marked as answer]
Same question not even two weeks old: #64697
